tests/test_fixed_grid_fourier_integral.py

```diff
@@ -0,0 +1,95 @@
+        0.5+3.333333333333333e-10j,
+        0.3817732906760362+0.3011686789397568j,
+        5.458434492865867e-10-8.378871808180589e-10j,
+        -9.92969320740405e-19-1.1837199021871074e-19j
+    ])
+
+    output_lambda = _lambda(input_x)
+
+    assert output_lambda == pytest.approx(expected_lambda, rel=comparison_tolerance, abs=comparison_tolerance)
+
+def test_phi_and_psi():
+    """Test _phi_and_psi against calculation using the analytical formula using 200 decimal digit precision with mpmath."""
+    input_x = np.array([1e-9, 1, 1e9, 1e18], dtype=np.float64)
+
+    # Set tolerance to 10*machine precision
+    comparison_tolerance = 10 * np.finfo(input_x.dtype).eps
+
+    # Expected outputs generated with mpmath at 200 decimal digit precision
+    expected_phi = np.array([
+        0.5 + 3.5e-10j,
+        0.37392456407295215539 + 0.31553567661778005804j,
+        5.4584344944869956752e-10 - 8.3788718136390234542e-10j,
+        -9.9296932074040507621e-19 - 1.1837199021871073261e-19j
+    ])
+
+    expected_psi = np.array([
+        -0.0833333333333333333 - 5e-11j,
+        -0.06739546857228461362 - 0.046145700566923663667j,
+        8.3788717918052853757e-19 + 5.4584345480024828642e-19j,
+        1.1837199021871073658e-37 - 9.9296932074040507374e-37j
+    ])
+
+    output_phi, output_psi = _phi_and_psi(input_x)
+
+    assert output_phi == pytest.approx(expected_phi, rel=comparison_tolerance, abs=comparison_tolerance)
+    assert output_psi == pytest.approx(expected_psi, rel=comparison_tolerance, abs=comparison_tolerance)
+
+
+@pytest.mark.parametrize("interpolation_mode", [mode.value for mode in InterpolationMode])
+@pytest.mark.parametrize(("input_func", "expected_transform"), [
+    ( lambda f: 1 / np.sqrt( 2 * np.pi * f),                        lambda t: np.sqrt(np.pi / (2 * t)) ),
+    ( lambda f: np.array([[1,2],[3,4]]) / np.sqrt( 2 * np.pi * f),  lambda t: np.array([[1,2],[3,4]]) * np.sqrt(np.pi / (2 * t)) ), # Test dimension handling
+    ( lambda f: 1 / (1 + (2 * np.pi * f)**2),                       lambda t: np.pi / 2 * np.exp(-t) ),
+])
+def test_fourier_integral_fixed_sampling_pchip(input_func: Callable[[ArrayLike], ArrayLike], expected_transform: Callable[[ArrayLike], ArrayLike], interpolation_mode: str):
+    """Test Fourier integral accuracy on function with an analytically known Fourier transform."""
+    input_frequencies = np.logspace(-10,20,1000)
+    input_times = np.logspace(-15, 0, 50)
+
+    input_func_arr = np.array([input_func(freq) for freq in input_frequencies])
+
+    output_transform_arr = fourier_integral_fixed_sampling(
+        times=input_times,
+        frequencies=input_frequencies,
+        func_values=input_func_arr,
+        inf_correction_term=True,
+        interpolation=interpolation_mode
+    )
+
+    expected_transform_arr = np.array([expected_transform(time) for time in input_times])
+
+    assert np.real(output_transform_arr) == pytest.approx(expected_transform_arr, rel=1e-3)
```

Comment:
I had to change the relative tolerance here to allow the tests to pass with the same frequency grid.

By increasing the number of frequencies in the frequency logspace (but keeping the same bounds) to 10000 or 50000, the maximum relative error (maximum over the time range) seems to stabilize around 4e-5 or lower for both methods and both frequency spacings.

At 10k frequencies, most points have error above 1e-6, while at 50k, only about the 10 largest times have errors above 1e-6.

Changing the frequency or time bounds, or adding more frequencies around features (which for these functions I suppose just means close to the origin), might help the accuracy. Let me know if I should investigate this more, and in that case what.

Reply:
> "I had to change the relative tolerance here to allow the tests to pass with the same frequency grid."

I changed it so that the testing tolerance depends on the interpolation type used. Now the pchip test is unchanged from last PR, but the tolerance is lowered for linear.